src/whisper.cpp

```diff
@@ -3667,6 +3667,9 @@ struct whisper_context * whisper_init_with_params_no_state(struct whisper_model_
     WHISPER_LOG_INFO("%s: gpu_device = %d\n", __func__, params.gpu_device);
     WHISPER_LOG_INFO("%s: dtw        = %d\n", __func__, params.dtw_token_timestamps);
 
+    // TODO: temporary call to force backend registry initialization
+    WHISPER_LOG_INFO("%s: backends   = %zu\n", __func__, ggml_backend_reg_count());
```

Comment:
I think all backends would currently fail because during loading the model we reference the backend buffer type:

https://github.com/ggerganov/whisper.cpp/blob/bc763c1d2c66bf5637fe3a644d0765a964410f13/src/whisper.cpp#L1802-L1807

But the `ggml_backend_registry` hasn't been initialized yet, before creating the first `whisper_state`:

https://github.com/ggerganov/whisper.cpp/blob/bc763c1d2c66bf5637fe3a644d0765a964410f13/src/whisper.cpp#L3326-L3334

I think after `whisper_default_buffer_type()` is reimplemented to use the backend registry, the problem will disappear and this workaround can be removed.